transport/src/main/java/io/netty/channel/AbstractChannel.java

```diff
@@ -46,13 +47,18 @@ public abstract class AbstractChannel extends DefaultAttributeMap implements Cha
 
     private static final InternalLogger logger = InternalLoggerFactory.getInstance(AbstractChannel.class);
 
+    private static final AtomicIntegerFieldUpdater<AbstractChannel> WRITABLE_UPDATER =
+            AtomicIntegerFieldUpdater.newUpdater(AbstractChannel.class, "writable");
+    private volatile int writable = 1;
```

Comment:
Nit, but I'd prefer if we could rely on zero-initialized to mean writable, so we don't have a volatile write in object initialization. Since this is really a boolean, it'd be good to have some constants to indicate the two values.